torch/csrc/utils/tensor_list.cpp

```diff
@@ -19,6 +19,7 @@ static PyObject* recursive_to_list(
   }
   auto n = sizes[dim];
   auto list = THPObjectPtr(PyList_New(n));
+  TORCH_CHECK(data, "tolist() shouldn't be called on a tensor with unallocated storage");
```

Comment:
if n is zero, will this spuriously throw even though a nullptr is ok?

Reply:
yeah you are right 
```
>>> torch.randn(2, 4, 0).tolist()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: tolist() shouldn't be called on a tensor with unallocated storage
```

Reply:
fixed it